pkg/virtctl/expose/expose_test.go

```diff
@@ -0,0 +1,36 @@
+package expose
+
+import (
+	"testing"
+
+	"github.com/spf13/pflag"
+
+	"kubevirt.io/kubevirt/pkg/kubecli"
+)
+
+// TODO: build a ginko test suite
+
+func Test_CommandCreation(t *testing.T) {
+	var flags pflag.FlagSet
+	clientConfig := kubecli.DefaultClientConfig(&flags)
+	cmd := NewExposeCommand(clientConfig)
+	if cmd == nil {
+		t.Error("'expose' command creation failure")
+	}
+	// TODO: verify content of command
+}
+
+func Test_Run(t *testing.T) {
+	var flags pflag.FlagSet
+	clientConfig := kubecli.DefaultClientConfig(&flags)
+	cmd := NewExposeCommand(clientConfig)
+	if cmd == nil {
+		t.Error("'expose' command creation failure")
+	}
+	// TODO: mock the client to not communicate with the server
```

Comment:
We have a mock-client which you can use for that purpose at https://github.com/kubevirt/kubevirt/blob/master/pkg/virt-controller/watch/vm_test.go#L145. It shows on how to set up our mock client and the fake client for k8s resources.

Reply:
Regarding functional tests, at https://github.com/kubevirt/kubevirt/blob/master/tests/ovm_test.go#L431 is an example on how one may test virtctl subcommands on a real cluster.